src/compiler/tracing.ts

```diff
@@ -162,11 +162,22 @@ namespace ts { // eslint-disable-line one-namespace-per-file
             performance.measure("Tracing", "beginTracing", "endTracing");
         }
 
-        function indexFromOne(lc: LineAndCharacter): LineAndCharacter {
-            return {
-                line: lc.line + 1,
-                character: lc.character + 1,
-            };
+        function getLocation(node: Node | undefined) {
+            const file = getSourceFileOfNode(node);
+            return !file
+                ? undefined
+                : {
+                    path: file.path,
+                    start: indexFromOne(getLineAndCharacterOfPosition(file, node!.pos)),
+                    end: indexFromOne(getLineAndCharacterOfPosition(file, node!.end)),
+                };
+
+            function indexFromOne(lc: LineAndCharacter): LineAndCharacter {
+                return {
+                    line: lc.line + 1,
+                    character: lc.character + 1,
+                };
+            }
         }

```

Comment:
Mainly moving the `getLineAndCharacterOfPosition` so it's mentioned once.  Feel free to ignore my flattening of the `return` and my dislike of vertical whitespace...

```suggestion
        function getLocation(node: Node | undefined) {
            const file = getSourceFileOfNode(node);
            if (!file) return;
            return {
                path: file.path,
                start: getPosFromOne(file, node!.pos),
                end: getPosFromOne(file, node!.end),
            };
            function getPosFromOne(file: SourceFile, pos: number): LineAndCharacter {
                const lc = getLineAndCharacterOfPosition(file, pos);
                return { line: lc.line + 1, character: lc.character + 1 };
            }
        }

```

Reply:
Thanks, but I don't find the improvement significant.